### PR TITLE
expression: Support GBK for builtin function AesEncrypt

### LINKED ISSUE #29168: Support GBK for builtin function `AesEncrypt`

### PR DESCRIPTION
**Pull request by jackwener**
### What problem does this PR solve?

Issue Number: close #29168

### What is changed

* Add charset convert for AesEncrypt args
* Add explain tests

### Check List

Tests

* [x] Unit test
* [x] Integration test

Side effects

* [x] None

Documentation

* [x] None

### Release note

```
 None
```

### CODE REVIEW

**Comment by unconsolable** [comment]
@jackwener Could you please generate corresponding `new_character_set_builtin.result` via `cmd/explaintest/run-test.sh`, and compare the result with MySQL? Thanks.

**Comment by xiongjiwei** [comment]
hi, is there any update?

**Comment by jackwener** [comment]
No, I find `shell` will build tidb and run, it is not a lightweight operation.

I will try to solve it again on the weekend

File: expression/builtin_encryption_vec.go [hunk 1]
```diff
@@ -67,9 +67,19 @@ func (b *builtinAesDecryptSig) vecEvalString(input *chunk.Chunk, result *chunk.C
 	isWarning := !b.ivRequired && len(b.args) == 3
 	isConstKey := b.args[1].ConstItem(b.ctx.GetSessionVars().StmtCtx)
 
-	var key []byte
```
Comment (zimulala) [inline expression/builtin_encryption_vec.go:70]: Have these two functions(`builtinAesDecryptSig` and `builtinAesDecryptIVSig`) been accidentally modified incorrectly?

**Comment by jackwener** [comment]
I should debug for finding it

**Pull request closed without merging**